File: 1784-check-if-binary-string-has-at-most-one-segment-of-ones/1784-check-if-binary-string-has-at-most-one-segment-of-ones.cpp
class Solution {
public:
    bool checkOnesSegment(string s) {
        int n = s.size();
        bool check = false;
        for(int i =  1; i < n; i++){
            if(check){
                if(s[i] == '1') return false;
            }
            else if(s[i] == '0') check = true;
        }
        return true;
    }
};

File: 1784-check-if-binary-string-has-at-most-one-segment-of-ones/1784-check-if-binary-string-has-at-most-one-segment-of-ones.py
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        n = len(s)
        check = False
        for i in range(n):
            if check:
                if s[i] == '1':
                    return False
            elif s[i] == '0':
                check = True

        return True